html_css/v2/main-course/custom-properties.md

```diff
@@ -0,0 +1,205 @@
+
+In the last example above, you may have noticed that we declared and then accessed our custom properties within the same declaration block. That's because the scope of a custom property is determined by the selector.
+
+~~~html
+<p class='cool-paragraph'>Lorem ipsum dolor sit amet.</p>
+
+<p class='boring-paragraph'>Lorem ipsum dolor sit amet.</p>
+~~~
+~~~css
+.cool-paragraph {
+  --main-bg: red;
+
+  background-color: var(--main-bg);
+}
+
+.boring-paragraph {
+  background-color: var(--main-bg);
+}
+~~~
+
+In the above example, only the element with the `cool-paragraph` class gets styled with a red background, while the `boring-paragraph` selector doesn't since it's not within the scope of where the custom property was declared. If you're familiar with how scope works in JavaScript, this sort of behavior should feel a little similar.
+
+That doesn't mean a custom property can *only* be accessed on the selector it was declared for, though. The scope includes the selector the custom property was declared for as well as the selector's descendants. Let's take a look at a modified version of the above example:
+
+~~~html
+<div class='cool-div'>
+  <p class='cool-paragraph'>Lorem ipsum dolor sit amet.</p>
+</div>
+~~~
+~~~css
+.cool-div {
+  --main-bg: red;
+}
+
+.cool-paragraph {
+  background-color: var(--main-bg);
+}
+~~~
+
+Even though we declared our custom property on our `.cool-div` selector, we can still access it on its descendant `.cool-paragraph` selector. Thanks, dad! Er, thanks `cool-div`!
+
+#### The `:root:` Selector
+
+While there may be times where you will want to limit the scope of a custom property, you may want to be able to use other custom properties on many, unrelated selectors. One work around would be declaring the same custom property on a bunch of selectors, but that defeats one of the purposes of using custom properties in the first place (the ease of changing multiple instances of a value at once).
+
+A better solution is declaring those custom properties on the `:root` selector, which is essentially the same thing as the `html` selector except it has a higher specificity.
+
+~~~html
+<p class='cool-paragraph'>Lorem ipsum dolor sit amet.</p>
+
+<p class='exciting-paragraph'>Lorem ipsum dolor sit amet.</p>
+~~~
+~~~css
+:root {
+  --main-color: red;
+}
+
+.cool-paragraph {
+  color: var(--main-color);
+}
+
+.exciting-paragraph {
+  background-color: var(--main-color);
+}
+~~~
+
+By declaring our custom property on the `:root` selector in the example above, we can access it on *any* other valid selector within our CSS file, since any other selector would be considered a descendant of the `:root` selector.
+
+### Themes
+
+Beyond allowing us to access custom properties more globally, the `:root` selector gives us one way to add themes to our pages. Keep in mind that we're not covering every way to add theming in this lesson, as we just want to introduce you to a simple way that you can get a lot of use out of.
+
```

Comment:
I would consider renaming, and reframing this section from "Themes" to "Case Study" or something. This way we don't need to explain to the reader that there are more ways to do themes.

Reply:
🤔 I like "themes" here actually... it's one of the big wins for custom properties.

if we need to be more specific lets do "creating themes with custom properties"

Reply:
Id happy with some more specificity here, I just don't want to imply that it's the ONLY thing you can do with this, or the ONLY way you can do themes. 

However, our readers aren't dumb, they can figure that out.

Reply:
True. I can change the section header to "Creating Themes with Custom Properties" like Cody said (or something similar),and just remove the second sentence.

"Beyond allowing us to access custom properties more globally, the `:root` selector gives us one way to add themes to our pages." Saying "one way to..." could definitely be enough to clue in readers that this isn't the only way to create themes. The second sentence is just me being very explicit (admittedly at times when really it's not needed).

Reply:
>  Saying "one way to..." could definitely be enough to clue
 100% this